packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -0,0 +1,332 @@
+// @ts-check
+import { E } from '@endo/eventual-send';
+import { Fail } from '@endo/errors';
+import {
+  assertPathSegment,
+  makeStorageNodeChild,
+} from '@agoric/internal/src/lib-chainStorage.js';
+import { reserveThenDeposit } from './utils.js';
+
+/** @import {EconomyBootstrapPowers} from './econ-behaviors.js' */
+/** @import {CommitteeElectorateCreatorFacet} from '@agoric/governance/src/committee.js'; */
+
+const trace = (...args) => console.log('GovReplaceCommiteeAndCharter', ...args);
+
+const traced = (label, x) => {
+  trace(label, x);
+  return x;
+};
+
+const { values } = Object;
+
+/** @type {<X, Y>(xs: X[], ys: Y[]) => [X, Y][]} */
+const zip = (xs, ys) => xs.map((x, i) => [x, ys[i]]);
+
+/** @type {(name: string) => string} */
+const sanitizePathSegment = name => {
+  const candidate = name.replace(/[ ,]/g, '_');
+  assertPathSegment(candidate);
+  return candidate;
+};
+
+/**
+ * Handles the configuration updates for high-priority senders list by adding or
+ * removing addresses.
+ *
+ * @param {EconomyBootstrapPowers} powers - The bootstrap powers required for
+ *   economic operations.
+ * @param {{
+ *   options: {
+ *     highPrioritySendersConfig: {
+ *       addressesToAdd: string[];
+ *       addressesToRemove: string[];
+ *     };
+ *   };
+ * }} config
+ *   - The configuration object containing lists of addresses to add or remove.
+ */
+const handlehighPrioritySendersList = async (
+  { consume: { highPrioritySendersManager: highPrioritySendersManagerP } },
+  { options: { highPrioritySendersConfig } },
+) => {
+  const HIGH_PRIORITY_SENDERS_NAMESPACE = 'economicCommittee';
+  const highPrioritySendersManager = await highPrioritySendersManagerP;
+
+  assert(
+    highPrioritySendersManager,
+    `highPriority SendersManager is not defined`,
+  );
+
+  if (!highPrioritySendersManager) {
+    Fail`highPrioritySendersManager is not defined`;
+  }
```

Comment:
1 of these should be enough.

```suggestion
  if (!highPrioritySendersManager) {
    throw assert.error(`highPrioritySendersManager is not defined`);
  }
```